datashader/transfer_functions/__init__.py

```diff
@@ -172,20 +172,22 @@ def eq_hist(data, mask=None, nbins=256*256):
 
     # Run more accurate value counting if data is of boolean or integer type
     # and unique value array is smaller than nbins.
-    if data2.dtype == bool or (np.issubdtype(data2.dtype, np.integer) and data2.max() < nbins):
+    if data2.dtype == bool or (np.issubdtype(data2.dtype, np.integer) and data2.ptp() < nbins):
         values, counts = np.unique(data2, return_counts=True)
-        vmin, vmax = values.min(), values.max()
+        vmin, vmax = values[0], values[-1]
```

Comment:
Values returned from `np.unique` are sorted so can grab the first and last values directly rather than walking the array twice to get the `min()` and `max()`.